src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -1109,6 +1162,9 @@ private static String addPrefix(String server) {
      * See http://msdn.microsoft.com/en-us/library/aa746475(VS.85).aspx for the syntax by example.
      * WANTED: The specification of the syntax.
      */
+    @Restricted(NoExternalUse.class)
+    @SuppressFBWarnings(value = "MS_SHOULD_BE_FINAL", 
+            justification = "Filter may be changed in the runtime by plugins, API backward compatibility")
```

Comment:
Sorry, it would be a formal backward compatibility issue. I won't do it without enough justification.